src/utils.ts

```diff
@@ -0,0 +1,11 @@
+import { ServerResponse } from 'node:http';
+
+import { green, red, yellow } from 'colorette';
+
+export const isWarnStatus = (res: ServerResponse) => res.statusCode >= 400 && res.statusCode < 499;
+export const isErrorStatus = (res: ServerResponse) => res.statusCode >= 500 && res.statusCode < 599;
```

Comment:
instead of writing `res.err || isErrorStatus(res)` in several spots, could we just make this method check for `res.err` as well? and perhaps rename them to something like `isErrorResponse()`